src/Components/ProductTable.jsx

```diff
@@ -1,53 +1,76 @@
-import React, { useEffect, useState } from "react";
-import ProductFilters from "./ProductFilters";
+import React, { useEffect, useState } from 'react';
+import ProductFilters from './ProductFilters';
 
 const getDefaultFilterOptions = () => {
   return {
     price: [
-      { minValue: 0, maxValue: 25, label: "$0 - $25", checked: false },
-      { minValue: 25, maxValue: 50, label: "$25 - $50", checked: false },
-      { minValue: 50, maxValue: 75, label: "$50 - $75", checked: false },
-      { minValue: 75, maxValue: Number.MAX_VALUE, label: "$75+", checked: false },
+      { minValue: 0, maxValue: 25, label: '$0 - $25', checked: false },
+      { minValue: 25, maxValue: 50, label: '$25 - $50', checked: false },
+      { minValue: 50, maxValue: 75, label: '$50 - $75', checked: false },
+      {
+        minValue: 75,
+        maxValue: Number.MAX_VALUE,
+        label: '$75+',
+        checked: false,
+      },
     ],
     color: [
-      { value: "beige", label: "Beige", checked: false },
-      { value: "green", label: "Green", checked: false },
-      { value: "white", label: "White", checked: false },
-      { value: "black", label: "Black", checked: false },
-      { value: "gray", label: "Gray", checked: false },
-      { value: "teal", label: "Teal", checked: false },
+      { value: 'beige', label: 'Beige', checked: false },
+      { value: 'green', label: 'Green', checked: false },
+      { value: 'white', label: 'White', checked: false },
+      { value: 'black', label: 'Black', checked: false },
+      { value: 'gray', label: 'Gray', checked: false },
+      { value: 'teal', label: 'Teal', checked: false },
     ],
   };
 };
 
 const getDefaultSortOptions = () => {
   return [
-    { name: "Price", current: false },
-    { name: "Newest", current: false },
+    { name: 'Price', current: false },
+    { name: 'Newest', current: false },
   ];
 };
 
 export default function ProductTable({ cart, updateCart }) {
   let [products, setProducts] = useState([]);
-
+  let [allproducts, setallProducts] = useState([]);
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
**praise:** The introduction of the allproducts state is a good way to store all fetched products, while the products state is used for filtered products. This separation ensures that the filter can be reset without fetching products again.

 ‏